### PR TITLE
Mark as unmaintained

### PR DESCRIPTION
**Pull request by andymckay**
Add in a notice about the upcoming lack of intended maintenance on this repository. Sadly we haven't got the people to maintain this repository and we'd like to make it clear to visitors to this repository about it's state. This process of marking it unmaintained only affects future updates. The Action will still continue work as it is in old workflows.

Once this lands, we'll also Archive this repository.

### CODE REVIEW

**Comment by jared-w** [comment]
When it's marked as unmaintained, can the README be updated to point to haskell-ci/setup as the maintained fork?

**Comment by andymckay** [comment]
>  When it's marked as unmaintained, can the README be updated to point to haskell-ci/setup as the maintained fork?

Sure 👍🏾 Added that in now.

**Comment by Anton-Latukha** [comment]
You always can:

1. Include active people in maintenance.
2. Repository can be transferred. So there is no need in forks, and users do not even notice that group owner changed, and would not need to do the migration.
3. Merge the changes faster, since they already made :), indeed if there was even a banner, I've spent some time yesterday, figuring out what is the error of 1.1, today - it does not even produce a syndrome.

**Comment by chshersh** [comment]
@jared-w Is the plan for `haskell-ci/setup` to become the official GitHub Action for Haskell instead of `actions/setup-haskell`?

**Comment by jared-w** [comment]
@chshersh Right now things are still slightly up in the air. Long story short is "essentially, yes, but with `haskell` being the org, not `haskell-CI`.

Originally, haskell-CI was where @hvr put the first fork of setup-actions and that's where I rewrote it. Now that it's in a place to become the official one, I thought about where it would make most sense for that to be and realized it really should be in the `haskell` github org rather than a haskell-CI one. This has the advantage of being in the "official" organization as well as the advantage of having an org with multiple members in it instead of just one.

Unfortunately I haven't been able to get a `haskell/actions` repo created yet. I've pinged someone about it and have been added to the org, but now I  need to go through the list of owners and see if someone has some free time to create the repo and set me as the maintainer/committer for it.

The final url,  I'm hoping, will be `haskell/actions/setup` which will open up the possibility of having more haskell-based github actions in a more conveniently discoverable place. I apologize for being a bit in the dark about this but I didn't want to prematurely announce things until I had at least been able to get into contact with a few people.

**Comment by andymckay** [comment]
Thanks for the ideas  @Anton-Latukha 

> You always can:
> 
>     1. Include active people into maintenance.

Unfortunately we can't give them rights to this repo because of certain organisational requirements.

>     2. Repository can be transferred. So there is no need in forks, and users do not even notice that group owner changed, and would not need to do the migration.

We could, except Actions do not recognise redirects. Also if we did there is a concern we'd be changing security boundaries significantly from a GitHub provided Action to a non-GitHub provided Action.

**Comment by jared-w** [comment]
@andymckay would it be possible to switch the README to point to [haskell/actions](https://github.com/haskell/actions) instead of haskell-CI/setup? After reflection, it made more sense to migrate the action over to the official Haskell organization rather than keep it in the haskell-ci org (which historically has not had the best discoverability).

Sorry for the extra trouble, but I'd really appreciate it :)

File: README.md [hunk 1]
```diff
@@ -1,5 +1,22 @@
 # setup-haskell
 
+**Please note:** This repository is currently unmaintained by a team of developers at GitHub. The 
+repository is here and you can use it as an example, or in Actions. However please be aware that 
+we are not going to  be updating issues or pull requests on this repository.
+
+**Maintained forks:**
+* [haskell-ci/setup](https://github.com/haskell-ci/setup)
```
Comment (andymckay) [inline README.md:8]:
```suggestion
* [haskell/actions](https://github.com/haskell/actions)
```

**Comment by andymckay** [comment]
Done

**Review by joshmgross — APPROVED** [review]

**Pull request merged**